op-conductor/flags/flags.go

```diff
@@ -0,0 +1,77 @@
+package flags
+
+import (
+	"fmt"
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/ethereum-optimism/optimism/op-node/flags"
+	opservice "github.com/ethereum-optimism/optimism/op-service"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	opmetrics "github.com/ethereum-optimism/optimism/op-service/metrics"
+	oppprof "github.com/ethereum-optimism/optimism/op-service/pprof"
+	oprpc "github.com/ethereum-optimism/optimism/op-service/rpc"
+)
+
+const EnvVarPrefix = "OP_CONDUCTOR"
+
+var (
+	ConsensusAddr = &cli.StringFlag{
+		Name:    "consensus.addr",
+		Usage:   "Address to listen for consensus connections",
+		EnvVars: opservice.PrefixEnvVar(EnvVarPrefix, "CONSENSUS_ADDR"),
+		Value:   "127.0.0.1",
+	}
+	ConsensusPort = &cli.IntFlag{
+		Name:    "consensus.port",
+		Usage:   "Port to listen for consensus connections",
+		EnvVars: opservice.PrefixEnvVar(EnvVarPrefix, "CONSENSUS_PORT"),
+		Value:   50050,
+	}
+	RaftServerID = &cli.StringFlag{
+		Name:    "raft.server.id",
+		Usage:   "Unique ID for this server used by raft consensus",
+		EnvVars: opservice.PrefixEnvVar(EnvVarPrefix, "RAFT_SERVER_ID"),
+	}
+	RaftStorageDIR = &cli.StringFlag{
+		Name:    "raft.storage.dir",
+		Usage:   "Directory to store raft data",
+		EnvVars: opservice.PrefixEnvVar(EnvVarPrefix, "RAFT_STORAGE_DIR"),
+	}
+	NodeRPC = &cli.StringFlag{
+		Name:    "node.rpc",
+		Usage:   "HTTP provider URL for op-node",
+		EnvVars: opservice.PrefixEnvVar(EnvVarPrefix, "NODE_RPC"),
+	}
+)
+
+var requiredFlags = []cli.Flag{
+	ConsensusAddr,
+	ConsensusPort,
+	RaftServerID,
+	RaftStorageDIR,
+	NodeRPC,
+}
+
+var optionalFlags = []cli.Flag{}
+
+func init() {
+	optionalFlags = append(optionalFlags, oprpc.CLIFlags(EnvVarPrefix)...)
+	optionalFlags = append(optionalFlags, oplog.CLIFlags(EnvVarPrefix)...)
+	optionalFlags = append(optionalFlags, opmetrics.CLIFlags(EnvVarPrefix)...)
+	optionalFlags = append(optionalFlags, oppprof.CLIFlags(EnvVarPrefix)...)
+	optionalFlags = append(optionalFlags, flags.RollupConfig)
```

Comment:
What would be the suggested way to read rollup config? should we reuse op-node flags or should we create new flags under `OP_CONDUCTOR`?

Reply:
Make new flags under OP_CONDUCTOR - we don't want users to have to define some env vars as `OP_NODE` but others as `OP_CONDUCTOR`.

Reply:
If there end up being lots of such flags, we can refactor to use a function that prefixes the env or something similar.

Reply:
It is indeed quite a lot of flags that are needed to load the correct rollup config. (Network, RollupConfig, BetaExtraNetworks, CanyonOverrideFlag, DeltaOverrideFlag)

I'm thinking to move flags to `op-service/flags`, under prefix `OP_SERVICE`, this way it could be used across services (it is currently used by op_program, bootnode, and now op-conductor)

One concern here is that this will ask all clients to rename their env vars which could be cumbersome, curious do we already have some kind of process / prior arts to this? (If not, I'm thinking we could read new parameter and fall back to old ones if users didn't change, and wait some time for the announcement to reach everybody)

Reply:
@mslipper reworked, wdyt?